### ISSUE TITLE
v2.8.0 - TypeScript: 'Invariant Violation' when updating array in object

### ISSUE THREAD

**Issue by dogan-acuris** [kolodny/immutability-helper#111]
Hi

We've tried to update to version 2.8.0 and we're getting an error thrown even with the simple test case below (running with Jest):

```simple-test.spec.ts
import update from 'immutability-helper'

interface TestObject {
  array: string[]
}

describe('Immutability helper', () => {
  it('boom!', () => {
    const object: TestObject = {
      array: ['a', 'b']
    }

    const result = update(object, {
      array: {
        $push: 'c'
      }
    })

    expect(result).toEqual({
      array: ['a', 'b', 'c']
    })
  })
})
```

```
Invariant Violation: update(): expected spec of $push to be an array; got c. Did you forget to wrap your parameter in an array?

      11 |     }
      12 | 
    > 13 |     const result = update(object, {
         |                    ^
      14 |       array: {
      15 |         $push: 'c'
      16 |       }

```

- Is this possibly a bug?
- There is a file named `typings-test.ts` at the root of this project. How is this test file being run as part of your build process?

**Comment by kolodny** [comment]
This should throw an error in all versions of the library ([it does in `v2.7.1`](https://codesandbox.io/s/znl7wnqr1x)). `2.8.0` only changed some typescript definitions. You need to change the argument to `$push` to be an array. So change this:

```js
      array: {
        $push: 'c'
      }

      // to this

      array: {
        $push: ['c']
      }
```

**Comment by dogan-acuris** [comment]
Hi kolodny, thanks for your quick reply

In that case there seems to be a bug in the type definition for $push added in 2.8.0.

See this pull request: https://github.com/kolodny/immutability-helper/pull/112

It could be beneficial to include simple TypeScript tests to the project to confirm that everything compiles OK at least.

**Comment by kolodny** [comment]
Closed via #113
